pkg/block/fetcher.go

```diff
@@ -401,14 +401,25 @@ func NewLabelShardedMetaFilter(relabelConfig []*relabel.Config) *LabelShardedMet
 	return &LabelShardedMetaFilter{relabelConfig: relabelConfig}
 }
 
+const blockIDLabel = "__block_id"
```

Comment:
Do we have a list of the various different magic label names we use? I'm worried about adding more of these in undocumented places. At one point we had something like `@thanos...` but I forget if we took it out

Reply:
https://github.com/thanos-io/thanos/blob/master/docs/proposals/201909_thanos_sharding.md#relabelling

Reply:
thanks :)

Reply:
The proposal was already `complete` but that label was previously unused in the repo. So I take it we simply reserved the labels

Reply:
yes we simply missed to implement it back then

Reply:
This was part of the https://thanos.io/proposals/201909_thanos_sharding.md/ and yes we need docs! :hugs:

Reply:
nit, `special label` rather than `special block`

Reply:
done